src/pseudo/PspUpf.jl

```diff
@@ -0,0 +1,200 @@
+           r2projsdr::Vector{Vector{Vector}}; identifier="")
+
+Construct a Unified Pseudopotential Format pseudopotential. Currently only
+norm-conserving potentials are supported.
+"""
+function PspUpf(Zion, lmax, rgrid::Vector{T}, drgrid, vloc, vloc_interp::IT, projs,
+                projs_interp, h, pswfcs, pswfc_occs, rhoatom, rvlocpzdr, r2projsdr;
+                identifier="") where {T, IT}
+    PspUpf{T,IT}(Zion, lmax, rgrid, drgrid, vloc, vloc_interp, projs, projs_interp, h,
+                 pswfcs, pswfc_occs, rhoatom, rvlocpzdr, r2projsdr, identifier)
+end
+
+function parse_upf_file(path; identifier=path)
+    pseudo = load_upf(path)
+
+    unsupported = []
+    pseudo["header"]["core_correction"]      && push!(unsupported, "non-lin. core correction")
+    pseudo["header"]["has_so"]               && push!(unsupported, "spin-orbit coupling")
+    pseudo["header"]["pseudo_type"] == "SL"  && push!(unsupported, "semilocal potential")
+    pseudo["header"]["pseudo_type"] == "US"  && push!(unsupported, "ultrasoft")
+    pseudo["header"]["pseudo_type"] == "PAW" && push!(unsupported, "plane-augmented wave")
+    pseudo["header"]["has_gipaw"]            && push!(unsupported, "gipaw data")
+    pseudo["header"]["pseudo_type"] == "1/r" && push!(unsupported, "Coulomb")
+    length(unsupported) > 0 && error("Pseudopotential contains the following unsupported" *
+                                     " features/quantities: $unsupported")
+
+    Zion        = Int(pseudo["header"]["z_valence"])     # Pseudo-ion charge
+    rgrid       = pseudo["radial_grid"]                  # Radial grid
+    drgrid      = pseudo["radial_grid_derivative"]       # Integration coeff.s
+    lmax        = pseudo["header"]["l_max"]              # Maximum angular momentum channel
+    vloc        = pseudo["local_potential"] ./ 2         # Local potential (Ry -> Ha)
+    vloc_interp = linear_interpolation((rgrid, ), vloc)  # Interp.or for the local potential
+
+    # Kleinman-Bylander β projectors projs[l+1][i][ir]
+    # NB: UPFs store rβ, not just β
+    projs = map(0:lmax) do l
+        betas_l = filter(beta -> beta["angular_momentum"] == l, pseudo["beta_projectors"])
+        map(beta -> beta["radial_function"] ./ 2, betas_l)  # Ry -> Ha
+    end
+    projs_interp = map(projs) do projs_l
+        map(projs_l) do proj
+            ir_cut = lastindex(proj)
+            linear_interpolation((rgrid[1:ir_cut], ), proj)
+        end
+    end
+
+    # β-projector coupling coefficients h[l+1][i,j] (also called e_{KB}, D_{ij})
+    h = Matrix[]
+    n = 1
+    for l = 0:lmax
+        nproj_l = length(projs[l+1])
+        Dij_l = pseudo["D_ion"][n:n+nproj_l-1, n:n+nproj_l-1] .* 2  # 1/Ry -> 1/Ha
+        push!(h, Dij_l)
+        n += nproj_l
+    end
+
+    # Pseudo-atomic wavefunctions
+    # Currently not used; can be used for initializing the starting wavefunction and as
+    # projectors for projected densities of states, projected wavefunctions, and DFT+U(+V).
```

Comment:
It's a nice comment that belongs to the struct rather than the internals of the constructor

Reply:
definitely good to have them parsed, it might be useful (eg @LaurentVidal95 for initializing Wannier?)

Reply:
If I understand correctly those are possible guesses for the initial projection of Bloch states for the Wannierization that mimic AOs ? Like sp like orbitals proposed in Quantum Espresso ?
For now the guess are random gaussian functions (which mostly works. Sometimes I get strange Wanniers but I only have to launch the Wannierization again to have what I need) so if the user can ask for something more specific that might be a nice addition !

Reply:
I think that Wannier90 does use these for the initial projection. Would be good for @LaurentVidal95 to chat with @qiaojunfeng